.github/workflows/main-ci.yml

```
@@ -0,0 +1,280 @@
+          DATABASE_URL: postgresql://postgres:postgres@localhost:5432/test_db
+          REDIS_URL: redis://localhost:6379/0
+        run: |
+          pytest -v -n auto \
+            --cov=. \
+            --cov-report=xml \
+            --cov-report=html \
+            --cov-report=term \
+            --tb=short
+
+      - name: Upload coverage
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.10)

[error] the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)